internal/venice-common/src/main/java/com/linkedin/venice/pubsub/api/PubSubPosition.java

```diff
@@ -0,0 +1,23 @@
+package com.linkedin.venice.pubsub.api;
+
+import com.linkedin.venice.pubsub.PubSubPositionUtils;
+
+
+/**
+ * Represents a position of a message in a partition of a topic.
+ */
+public interface PubSubPosition {
+  int comparePosition(PubSubPosition other);
+
+  long diff(PubSubPosition other);
```

Comment:
Is this only going to be supportable in Kafka? Is it viable to have this in the API?